### PR TITLE
POL-650 Make Org ID mandatory and account optional in DB

### PR DESCRIPTION
**Pull request by gwenneg**
Checked with `gabi` today:
| Query | Stage | Prod |
| - | - | - |
| select count(*) from policy where org_id is null | 0 | 0 |
| select count(*) from policies_history where org_id is null | 0 | 0 |

### CODE REVIEW

File: src/main/java/com/redhat/cloud/policies/app/auth/IncomingRequestFilter.java [hunk 1]
```diff
@@ -136,9 +136,10 @@ private XRhIdentity determineXRhIdentity(String xrhid_header) {
         // Basic sanity check
         XRhIdentity rhIdentity = xrhid.get();
         if (rhIdentity.getUsername() == null || rhIdentity.getUsername().isEmpty() ||
-                rhIdentity.identity.accountNumber == null || rhIdentity.identity.accountNumber.isEmpty()
+                rhIdentity.identity.accountNumber == null || rhIdentity.identity.accountNumber.isEmpty() ||
```
Comment (josejulio) [inline src/main/java/com/redhat/cloud/policies/app/auth/IncomingRequestFilter.java:139]: Should we remove this check? IIRC some users won't have an account id

Comment (gwenneg) [reply src/main/java/com/redhat/cloud/policies/app/auth/IncomingRequestFilter.java]: Done

File: src/main/resources/db/migration/V21__POL-650_org_id_constraints.sql [hunk 2]
```diff
@@ -0,0 +1,13 @@
+DROP INDEX pol_cus_nam_idx;
+
+ALTER TABLE policy
+    DROP CONSTRAINT policy_pkey,
+    ALTER COLUMN customerid DROP NOT NULL,
```
Comment (josejulio) [inline src/main/resources/db/migration/V21__POL-650_org_id_constraints.sql:5]: isn't the  customer_id the account_id field? Are we removing it?

File: src/main/resources/db/migration/V21__POL-650_org_id_constraints.sql [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    ALTER COLUMN org_id SET NOT NULL,
+    ADD CONSTRAINT policy_pkey PRIMARY KEY (org_id, id);
+
+CREATE UNIQUE INDEX policy_org_id_name_idx ON policy (org_id, name);
+
+ALTER TABLE policies_history
+    ALTER COLUMN tenant_id DROP NOT NULL,
```
Comment (josejulio) [inline src/main/resources/db/migration/V21__POL-650_org_id_constraints.sql:12]: same as above

File: src/main/resources/db/migration/V21__POL-650_org_id_constraints.sql (hunk 2 above)
Comment (gwenneg) [reply src/main/resources/db/migration/V21__POL-650_org_id_constraints.sql:5]: Yes, this is where the account ID is stored, but this script only removes a `NOT NULL` constraint, not the field itself.

[the remark this answers is not in the document]
Comment (josejulio) [reply src/main/resources/db/migration/V21__POL-650_org_id_constraints.sql:5]: ah - nevermind, its removing the NOT NULL

File: src/main/resources/db/migration/V21__POL-650_org_id_constraints.sql (hunk 3 above)
Comment (josejulio) [reply src/main/resources/db/migration/V21__POL-650_org_id_constraints.sql:12]: nevermind this one too :D

**Review by josejulio — APPROVED** [review]
lgtm

**Pull request merged**